WordPress/Classes/Networking/MediaServiceRemoteREST.m

```diff
@@ -187,6 +187,29 @@ - (void)updateMedia:(RemoteMedia *)media
            }];
 }
 
+- (void)deleteMedia:(RemoteMedia *)media
+            success:(void (^)())success
+            failure:(void (^)(NSError *))failure
+{
+    NSParameterAssert([media isKindOfClass:[RemoteMedia class]]);
+
+    NSString *path = [NSString stringWithFormat:@"sites/%@/media/%@/delete", self.siteID, media.mediaID];
+    NSString *requestUrl = [self pathForEndpoint:path
+                                     withVersion:ServiceRemoteWordPressComRESTApiVersion_1_1];
+
+    [self.wordPressComRestApi POST:requestUrl
+                        parameters:nil
+                           success:^(id responseObject, NSHTTPURLResponse *response) {
```

Comment:
While a bit painful, probably best go ahead and knock it out now with something like https://github.com/wordpress-mobile/WordPress-iOS/blob/develop/WordPress/Classes/Services/MediaService.m#L339 and a message of "Something went wrong, please try again" or the like.